client/core/core.go

```diff
@@ -1255,6 +1260,58 @@ func (c *Core) wallet(assetID uint32) (*xcWallet, bool) {
 	return w, found
 }
 
+// GetCachedPassword takes the mapping string passed to CacheAppPassword and
+// the key returned from it to lookup and decrypt the app password.
+func (c *Core) GetCachedPassword(mapping string, key []byte) ([]byte, error) {
+	_, pushes, err := encode.DecodeBlob(key)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding blob: %v", err)
+	}
+
+	key, deserializedCrypter := pushes[0], pushes[1]
```

Comment:
The name `key` is used to represent two different things here. Maybe the method argument can be named `keyBlob` or something.